docs/provider/provider-tls.md

```diff
@@ -0,0 +1,361 @@
+<br />
+
+### 💻 Validate Certificate
+
+Let's make sure your certificate successfully installed! ✅ Input the following command:
+
+```sudo certbot certificates```
+
+Keep track of your output. If your certificate generation was successful, it should look as following:
+
+```
+Found the following certs:
+  Certificate Name: your-site.com
+    Domains: your-site.com eth.your-site.com lava.your-site.com
+    Expiry Date: 2023-11-07 14:37:29+00:00 (VALID: 84 days)
+    Certificate Path: /etc/letsencrypt/live/your-site.com/fullchain.pem
+    Private Key Path: /etc/letsencrypt/live/your-site.com/privkey.pem
+
+```
+
+You'll need both `Certificate Path` and `Private Key Path` for your next step.
+
+<br />
+
+### 🗃️ Add an Nginx Config for Each Domain
+
+Lava recommends running each chain under a separate provider process. This will separate error logs and protect against complete provider failure in the case of a problematic provider process. The first step of this is to create different nginx routes for each chain.
+
+For each chain that you want to support, you will need to create a separate `nginx` config file. 
+`cd` into `/etc/sites-available` and create a `server` file for each chain. You will need to select an open port for each chain. `Nginx` will use these config files to create your routes.
+
+<Tabs>
+<TabItem value="eth_ex" label="eth_server">
+
+🟢 ```sudo nano eth_server``` 
+
+
+```
+server {
+    listen 443 ssl http2;
+    server_name eth.your-site.com;
+
+    ssl_certificate /etc/letsencrypt/live/your-site.com/fullchain.pem;
+    ssl_certificate_key /etc/letsencrypt/live/your-site.com/privkey.pem;
+    error_log /var/log/nginx/debug.log debug;
+
+    location / {
+        proxy_pass http://127.0.0.1:2223;
+        grpc_pass 127.0.0.1:2223;
+    }
+}
+```
+
+</TabItem>
+<TabItem value="lav_ex" label="lava_server">
+
+🟢 ```sudo nano lava_server```
+
+```
+server {
+    listen 443 ssl http2;
+    server_name lava.your-site.com;
+
+    ssl_certificate /etc/letsencrypt/live/your-site.com/fullchain.pem;
+    ssl_certificate_key /etc/letsencrypt/live/your-site.com/privkey.pem;
+    error_log /var/log/nginx/debug.log debug;
+
+    location / {
+        proxy_pass http://127.0.0.1:2224;
+        grpc_pass 127.0.0.1:2224;
+    }
+}
+```
+
+
+</TabItem>
+</Tabs>
+
+:::tip
+
+The above examples use ports `2223` and `2224`, respectively. You can choose any port that works for your purposes. 
+
+:::
+
+<br />
+
+
+### ♻️ Restart Nginx
+
+You will need to refresh the Nginx server:
+
+```bash
+sudo systemctl nginx restart
+```
+<br />
+
+### 🧪 Test Nginx Configuration
+
+Now, ensure that your `nginx` setup is working! ✅
+
+```bash
+sudo nginx -t
+```
+
+🖳 Expected Output:
+```
+nginx: the configuration file /etc/nginx/nginx.conf syntax is ok
+nginx: configuration file /etc/nginx/nginx.conf test is successful
+```
+
+<br />
+
+The syntax on your files must be precise. Misplaced or invisible characters or inconsistent indentation can cause errors.
+
```

Comment:
What do you think about moving this section to before restarting Nginx? If there's an error/typo in the Nginx configuration, the previous section will error. Checking the configuration before restarting/reloading will allow people to fix mistakes before moving to the next section.

Also, Nginx does not care about indentation. That won't give an error. Maybe you meant "inconsistent curly braces"?

Reply:
I think it's a good idea to move it. I will change and link to the commit here!

My mistake on the indentation - I was thinking of the yaml files that are fed to start the rpcprovider process. I think that whole comment was meant to be applied to them. I will move to appropriate place.

Reply:
✅ Fixed 
Commit [d858ca0e5c0384a43dca7c9f45a9fba7c7439a2d](https://github.com/lavanet/docs/pull/151/commits/d858ca0e5c0384a43dca7c9f45a9fba7c7439a2d)